Core/Core/Planner/CalendarEvent/ViewModel/EditCustomFrequencyViewModel.swift

```diff
@@ -0,0 +1,276 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import Combine
+
+final class EditCustomFrequencyViewModel: ObservableObject {
+
+    // MARK: - Page Setup
+
+    let pageTitle = String(localized: "Custom Frequency", bundle: .core)
+    let doneButtonTitle: String = String(localized: "Done", bundle: .core)
+    let pageViewEvent = ScreenViewTrackingParameters(eventName: "/calendar/new/frequency/custom")
+    let screenConfig = InstUI.BaseScreenConfig(refreshable: false)
+
+    // MARK: - Data
+    let proposedDate: Date
+
+    // MARK: - Actions
+    let didTapDone = PassthroughSubject<WeakViewController, Never>()
+
+    // MARK: - Inputs / Outputs
+
+    @Published private(set) var state: InstUI.ScreenState = .data
+
+    @Published var frequency: RecurrenceFrequency = .daily
+    @Published var interval = FrequencyInterval(value: 1)
+
+    @Published var endMode: EndMode?
+    @Published var endDate: Date? = Clock.now
+    @Published var occurrenceCount: Int
+
+    /// Specific to Weekly
+    @Published var daysOfTheWeek: [Weekday] = []
+
+    /// Specific to Monthly
+    @Published var dayOfMonth: DayOfMonth?
```

Comment:
We seems to have missed this, but this should default to the currently selected day of month.
It was already in the figma, but now it's [highlighted](https://www.figma.com/design/asf3Fh7xqLsFpVOJND5QLd/%E2%9C%85-%5BDelivered%5D-(Canvas-Teacher)-Calendar-V01--Marci?node-id=8012-27702&t=wbEuSvUtdNY7YJCQ-4).
This can't be unselected, so it may make sense to make it non-optional, if it doesn't complicate init too much